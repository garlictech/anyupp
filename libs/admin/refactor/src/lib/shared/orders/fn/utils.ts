import { CrudSdk } from '@bgap/crud-gql/api';

export interface OrderHandlerDeps {
  crudSdk: CrudSdk;
  timestamp: () => number;
}
