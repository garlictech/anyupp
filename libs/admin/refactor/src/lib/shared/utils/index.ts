import { ImageCompressorService } from './services/image-compressor/image-compressor.service';
import { QrGeneratorService } from './services/qr-generator/qr-generator.service';
import { ToasterService } from './services/toaster/toaster.service';

export * from './const';
export * from './enums';
export * from './fn';
export { ImageCompressorService, ToasterService, QrGeneratorService };
