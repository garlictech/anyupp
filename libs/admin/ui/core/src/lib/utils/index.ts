import { AnalyticsService } from './analytics.service';
import { LayoutService } from './layout.service';
import { StateService } from './state.service';
import { CustomNbIconLibraries } from './custom-icon-libraries.service';

export { LayoutService, AnalyticsService, StateService, CustomNbIconLibraries };
