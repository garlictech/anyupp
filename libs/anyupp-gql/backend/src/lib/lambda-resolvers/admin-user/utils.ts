import { CognitoIdentityServiceProvider, DynamoDB } from 'aws-sdk';
import * as CrudApi from '@bgap/crud-gql/api';
import { from } from 'rxjs';
import { map, tap } from 'rxjs/operators';

export interface AdminUserResolverDeps {
  userPoolId: string;
  cognitoidentityserviceprovider: CognitoIdentityServiceProvider;
  userNameGenerator: () => string;
  docClient: DynamoDB.DocumentClient;
  adminUserTableName: string;
}

export const deleteAdminUserFromTable =
  (deps: AdminUserResolverDeps) =>
  (username: string): ReturnType<CrudApi.CrudSdk['DeleteAdminUser']> =>
    from(
      deps.docClient
        .delete({
          TableName: deps.adminUserTableName,
          Key: {
            id: username,
          },
        })
        .promise(),
    ).pipe(map(res => res.Attributes as CrudApi.Maybe<CrudApi.AdminUser>));
