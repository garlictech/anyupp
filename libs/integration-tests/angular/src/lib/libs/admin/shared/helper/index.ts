import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
