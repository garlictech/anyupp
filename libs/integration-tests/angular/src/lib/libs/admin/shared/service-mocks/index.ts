export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
