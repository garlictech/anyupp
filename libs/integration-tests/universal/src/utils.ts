import { expect } from '@jest/globals';

export const ES_DELAY = 3000; //ms
export const DYNAMO_DELAY = 1000; //ms

export const dateMatcher = {
  createdAt: expect.any(String),
  updatedAt: expect.any(String),
};
