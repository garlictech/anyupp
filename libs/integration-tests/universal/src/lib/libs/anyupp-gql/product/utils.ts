import * as CrudApi from '@bgap/crud-gql/api';
import { throwError } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { throwIfEmptyValue } from '@bgap/shared/utils';

export const getChainProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetChainProduct({ id: productId }).pipe(
    throwIfEmptyValue<CrudApi.ChainProduct>(),
    catchError(err => {
      return throwError('ChainProdcuct is missing');
    }),
  );
};

export const getGroupProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetGroupProduct({ id: productId }).pipe(
    throwIfEmptyValue<CrudApi.GroupProduct>(),
    catchError(err => {
      return throwError('GroupProdcuct is missing');
    }),
  );
};

export const getUnitProduct = (sdk: CrudApi.CrudSdk, productId: string) => {
  return sdk.GetUnitProduct({ id: productId }).pipe(
    throwIfEmptyValue<CrudApi.UnitProduct>(),
    catchError(err => {
      return throwError('UnitProdcuct is missing');
    }),
  );
};
