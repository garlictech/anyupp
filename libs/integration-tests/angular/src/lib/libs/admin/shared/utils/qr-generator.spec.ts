import { getPDF, getQR } from '@bgap/admin/shared/utils';

describe('QrGenerator', () => {
  it('getQR should create SVG', async () => {
    expect(await getQR('unitId', 'tableId', 'seatId')).toMatchSnapshot();
  });

  it('getPDF should create QR PDF', async () => {
    expect(await getPDF('mock_QR_SVG_content')).toMatchSnapshot();
  });
});
