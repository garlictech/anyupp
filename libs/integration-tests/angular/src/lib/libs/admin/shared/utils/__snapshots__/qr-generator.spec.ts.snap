// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrGenerator getQR should create SVG 1`] = `
"                                         
                                         
    █▀▀▀▀▀█ ▀█▄▀▄ ▀▀ █▀▄ █▄▀  █▀▀▀▀▀█    
    █ ███ █ ▄ ▄▄▄▀▄█ ▄▄██▄▀█  █ ███ █    
    █ ▀▀▀ █ █ ▀█▄ ▀ ▄▀▀█▀█▄ █ █ ▀▀▀ █    
    ▀▀▀▀▀▀▀ █ █▄▀ █▄█▄█ ▀▄█▄█ ▀▀▀▀▀▀▀    
    █ ▄▄▀ ▀█▀ ▄▀█▄ ▄▄▀█▀▀▀█▄▀█▀▀▀█▄▄▀    
    █▀▀▄ ▀▀▀█▀ ▀▀ ▄█▄█▄▄▀█▀▀▄▄▄  ▀ ▀▄    
    ▀▄▄▄█▄▀▄ ▀█▄▀█▀▀▄▄▀█▀▄▄█▀▄▀█▀█ ▀▀    
      █▀▄▀▀ █  █▄▀▄ ▀▄██▄█▀█▀██  ▀ ▀     
    ▀█▄▄██▀▄  ▄  █▀▀▄█  ▀████ ▀ ▀▄▄█     
      ▄▄  ▀▀ ▄█▀▀▀ █▀ ▀▄▄ ▄▀██ █ ▀ █     
    ▄▄▀▀█▀▀ █ ▀▄ ▄▄ ▀ ▄▄▄ ▀ ▀▄▀ ▀█▄█▀    
        █▄▀ ▄▀▀ ▄  █▄██ ▀█ █▄█▄▀█  ▀     
    ▀▀▀▀ ▀▀ █▄ ▄█▄ ▄█▀▄▀█▀█▄█▀▀▀█ █▄     
    █▀▀▀▀▀█ ▀ █▄▀▀ █▄█▄  █▀ █ ▀ █▀ ▀     
    █ ███ █ ▀ ▄▄█▄▀▀▄▄▀█▀ ▄▄▀▀███▀ ▄▀    
    █ ▀▀▀ █  ▀▄██ █ ▀▄▀█▄██ ▄▀▀▀▀        
    ▀▀▀▀▀▀▀ ▀ ▀▀▀ ▀▀ ▀   ▀▀▀▀ ▀     ▀    
                                         
                                         "
`;
